### ISSUE TITLE
gatsby-transformer-remark: Throws an error when `constructor` is used as a field value

### ISSUE THREAD

**Issue by stephencweiss** [gatsbyjs/gatsby#22508]
## Description

I use a tag system for my website's blog. I recently wrote a post about instantiating classes using a `constructor` function -- I added this as a tag to the post and `gastsby-node` broke.

So too did my graphql query on my `tags` page.

I _think_ this is likely because of some conversion where the runtime is viewing `constructor` as a reserved word... though this is the _only_ reserved word I've come across with this behavior. 

### Steps to reproduce

Minimal Reproduction: 
- Repo: https://github.com/stephencweiss/gatsby-bug-constructor-tag
- Install deps and start repo `gatsby develop` -- you should see an error
- Remove line 3 in `./content/boo.md` and re-run -- you should see the error disappear

### Expected result

I should be able to have a tag of any valid string and not run into the situation where it breaks the build of my site. 

### Actual result

```shell
$ yarn build

 ERROR #11321  PLUGIN

"gatsby-node.js" threw an error while running the createPages lifecycle:

(acc[key] || []).concat is not a function

failed createPages - 0.064s

 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

(acc[key] || []).concat is not a function

   1 | query usersstephencodeblogsrcpagestagsJsx3450235017 {
   2 |   site {
   3 |     siteMetadata {
   4 |       title
   5 |     }
   6 |   }
   7 |   allMarkdownRemark(limit: 2000) {
>  8 |     group(field: frontmatter___tags) {
     |     ^
   9 |       fieldValue
  10 |       totalCount
  11 |     }
  12 |   }
  13 | }
  14 |

File path: /Users/stephen/code/blog/src/pages/tags.jsx
Url path: /tags/
Plugin: none
```

### Environment

```
  System:
    OS: macOS 10.15.3
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.16.1 - ~/.nvm/versions/node/v12.16.1/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.13.4 - ~/.nvm/versions/node/v12.16.1/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 80.0.3987.149
    Safari: 13.0.5
  npmPackages:
    gatsby: ^2.19.45 => 2.19.45 
    gatsby-source-filesystem: ^2.2.2 => 2.2.2 
    gatsby-transformer-remark: ^2.7.1 => 2.7.1 
```

**Comment by vladar** [comment]
The error you see is a runtime GraphQL error with `group` by `frontmatter___tags`. If you change grouping to something else - it doesn't break anymore. Still looking into why this happens with `frontmatter___tags`

**Comment by vladar** [comment]
Oh, I see what you mean now. Here is what happens:

```js
const a = {}
a[`constructor`] // expected: undefined, got function
```

Here is a PR with a fix: #22603
